src/server/utils/requester/requester.ts

```diff
@@ -30,35 +30,40 @@ export interface ProperRequesterOptions {
   druidRequestDecorator?: DruidRequestDecorator;
 }
 
-function httpToPlywoodProtocol(protocol: string): "plain" | "tls" {
+type PlywoodProtocol = "plain" | "tls";
+
+function httpToPlywoodProtocol(protocol: string): PlywoodProtocol {
   if (protocol === "https:") return "tls";
   return "plain";
 }
 
-function defaultPorts(url: URL): URL {
-  if (url.port) return url;
-  if (url.protocol === "http:") {
-    const newUrl = new URL(url.toString());
-    newUrl.port = "80";
-    return newUrl;
+function defaultPort(protocol: string): number {
+  switch (protocol) {
+    case "http:":
+      return 80;
+    case "https:":
+      return 443;
+    default:
+      throw new Error(`Unsupported protocol: ${protocol}`);
   }
-  if (url.protocol === "https:") {
-    const newUrl = new URL(url.toString());
-    newUrl.port = "443";
-    return newUrl;
+}
+
+function getHostAndProtocol(url: URL): { host: string, protocol: PlywoodProtocol } {
+  const { protocol, port, host } = url;
+  const plywoodProtocol = httpToPlywoodProtocol(protocol);
+  if (port) {
+    return { host, protocol: plywoodProtocol };
```

Comment:
magic, if port is defined - do not use it

Reply:
It's part of the `host`. But we can do better, and use `hostname` instead!